src/Stratis.FederatedPeg.Features.FederationGateway/FederatedPegBlockDefinition.cs

```diff
@@ -43,13 +43,11 @@ public class FederatedPegBlockDefinition : SmartContractPoABlockDefinition
 
         public override BlockTemplate Build(ChainedHeader chainTip, Script scriptPubKey)
         {
-            var rewardScript = chainTip.Height == this.Network.Consensus.PremineHeight 
+            var rewardScript = chainTip.Height + 1 == this.Network.Consensus.PremineHeight 
```

Comment:
Perhaps add parentheses:
`var rewardScript = (chainTip.Height + 1) == this.Network.Consensus.PremineHeight`